File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/clinical/infrastructure/events/InMemoryEventPublisher.java
package io.zenwave360.example.clinicaltool.modules.clinical.infrastructure.events;

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.events.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.events.EventPublisher;

import java.util.ArrayList;
import java.util.List;

public class InMemoryEventPublisher implements EventPublisher {

    private final List<Object> events = new ArrayList<>();

    public List<Object> getEvents() {
        return events;
    }
    public void onDoctorCreated(DoctorCreated event) {
        events.add(event);
    }
    public void onPatientCreated(PatientCreated event) {
        events.add(event);
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/example/asyncapi/provider/clinicaltool/modules/clinical/infrastructure/events/InMemoryEventPublisher.kt
package io.example.asyncapi.provider.clinicaltool.modules.clinical.infrastructure.events

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.events.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.events.EventPublisher

class InMemoryEventPublisher : EventPublisher {

    private val events = mutableListOf<Any>()

    fun getEvents(): MutableList<Any> {
        return events
    }
    override fun onDoctorCreated(event: DoctorCreated) {
        events.add(event)
    }
    override fun onPatientCreated(event: PatientCreated) {
        events.add(event)
    }
}
